src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1572 @@
+  static constexpr t8_element_coord
+  element_get_len (const t8_element_level level) noexcept
+  {
+    return 1 << (T8_ELEMENT_MAXLEVEL[TEclass] - (level));
+  }
+
+  static t8_cube_id
+  compute_cubeid (const t8_standalone_element<TEclass> *elem, const t8_element_level level) noexcept
+  {
+    t8_cube_id cube_id = 0;
+
+    T8_ASSERT (0 <= elem->level && elem->level <= T8_ELEMENT_MAXLEVEL[TEclass]);
+    const t8_element_coord h = element_get_len (level);
+
+    /* The cube id of the root element is 0.*/
+    if (level == 0) {
+      return 0;
+    }
+    for (int i = 0; i < T8_ELEMENT_DIM[TEclass]; i++) {
+      cube_id |= ((elem->coords[i] & h) ? 1 << i : 0);
+    }
+    return cube_id;
+  }
+
+  /**
+ * Compute the ancestor of \a el at a given level via the equation properties
+ * 
+ * \param[in] elem      Input element
+ * \param[in] level     Level of the ancestor to compute
+ * \param[in, out] and  Allocated element that will be filled with the data of the ancestor.
+ */
+  static constexpr void
+  element_get_ancestor (const t8_standalone_element<TEclass> *elem, const t8_element_level level,
+                        t8_standalone_element<TEclass> *ancestor) noexcept
+  {
+    T8_ASSERT (0 <= level && level <= elem->level);
+    if (elem != ancestor) {
+      element_copy ((const t8_element_t *) elem, (t8_element_t *) ancestor);
+    }
+    if (elem->level == level) {
+      return;
+    }
+
+    /* Set type */
+    if constexpr (T8_ELEMENT_NUM_EQUATIONS[TEclass]) {
+      SC_ABORT ("Only implemented for hypercubes.\n");
+    }
+
+    /* The coordinates and the type of the ancestor are defined by the level. */
+    element_cut_coordinates (ancestor, T8_ELEMENT_MAXLEVEL[TEclass] - level);
+
+    ancestor->level = level;
+  }
+
+  static constexpr int
+  element_get_cube_nca_level (const t8_standalone_element<TEclass> *elem1,
+                              const t8_standalone_element<TEclass> *elem2) noexcept
+  {
+    /* XOR all coordinates. The number of zeros on the left determines the level needed, so that the coordinates equal. 
+    OR over all these bit representations. The number of zeros on the left in this new number equals the coarses of all of these levels. 
+    Therefore this is the level needed so that all coordinates equal.*/
+    t8_element_coord maxexclor = 0;
+
+    for (int idim = 0; idim < T8_ELEMENT_DIM[TEclass]; idim++) {
+      maxexclor |= (elem1->coords[idim] ^ elem2->coords[idim]);
+    }
+
+    const int num_zeros = number_of_leading_zeros (maxexclor);
+    /* If one element already is the ancestor of the other element num_zeros evaluates to maxlevel, in that case return the coarser of both levels*/
+    return SC_MIN (num_zeros, (int) SC_MIN (elem1->level, elem2->level));
+  }
+
+  static constexpr int
+  number_of_leading_zeros (const t8_element_coord maxexclor) noexcept
+  {
+    const int num_of_active_bits_used = SC_LOG2_32 (maxexclor) + 1;
```

Comment:
rename maxexclor to something more generic, as this function does not care about where the input comes from.